pmlc/ast/ast_ops.cc

```diff
@@ -73,13 +78,19 @@ struct GatherOp : Intrinsic {
     }
     auto tensor = shapes[0];
     auto idxs = shapes[1];
+    if (isDataTypeFloat(idxs.elementType)) {
+      if (!isDataTypeFloat(tensor.elementType)) {
```

Comment:
Could you tighten this to a single if statement?

Reply:
Fixed